.scripts/ls.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env -S uv run --script --no-project
+
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Output the packages in the repository as a json list.
+
+See the the commandline help for more options.
+"""
+
+from __future__ import annotations
+
+import argparse
+import contextlib
+import dataclasses
+import io
+import json
+import logging
+import pathlib
+import subprocess
+import tarfile
+import tempfile
+
+_REPO_ROOT = pathlib.Path(__file__).parent.parent
+_INTERFACES = _REPO_ROOT / 'interfaces'
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(str(pathlib.Path(__file__).relative_to(_REPO_ROOT)))
+
+
+@dataclasses.dataclass
+class _Args:
+    kind: str
+    refs: tuple[str, str | None] | None
+    include_examples: bool
+    include_placeholders: bool
+    only_if_version_changed: bool
+
+    @classmethod
+    def from_cli(cls) -> _Args:
+        parser = argparse.ArgumentParser()
+        parser.add_argument('kind', choices=('packages', 'interfaces'))
+        parser.add_argument('old_ref', nargs='?')
+        parser.add_argument('new_ref', nargs='?')
+        parser.add_argument('--exclude-examples', action='store_true')
+        parser.add_argument('--exclude-placeholders', action='store_true')
+        parser.add_argument('--only-if-version-changed', action='store_true')
+        args = parser.parse_args()
+        return cls(
+            kind=args.kind,
+            refs=(args.old_ref, args.new_ref) if args.old_ref is not None else None,
+            include_examples=not args.exclude_examples,
+            include_placeholders=not args.exclude_placeholders,
+            only_if_version_changed=args.only_if_version_changed,
+        )
+
+
+def _ls(args: _Args) -> list[str]:
+    include: list[str] = []
+    if args.include_examples:
+        include.extend(('.example', '.tutorial'))
+    if args.include_placeholders:
+        include.append('.package')
+    # collect packages or interfaces
```

Comment:
Feel free to reject if this doesn't align with the style for the repo.

```suggestion
    # Collect packages or interfaces.
```